test/models/hostgroup_test.rb

```diff
@@ -389,6 +389,45 @@ class HostgroupTest < ActiveSupport::TestCase
     hostgroup.to_param
   end
 
+  test 'with both subnet and subnet6 should be valid if VLAN ID is consistent between subnets' do
+    existing_domain = Domain.first
+    subnet = FactoryBot.create(:subnet_ipv4, :domains => [existing_domain], :vlanid => 14)
+    subnet6 = FactoryBot.create(:subnet_ipv6, :domains => [existing_domain], :vlanid => 14)
+    hostgroup = FactoryBot.build(:hostgroup, :subnet => subnet, :subnet6 => subnet6)
+    assert hostgroup.valid?
```

Comment:
we have a nice helper for this that provides nicer debugging output if the model is not valid.

```ruby
assert_valid hostgroup
```